### PR TITLE
Introduce namelist option for thermal conductivity of snow: Jordan vs. Sturm

### PR DESCRIPTION
**Pull request by AdrienDams**
### Description of changes

We have replaced the snow thermal conductivity calculations from Jordan (1991) with [Sturm et al. (1997)](https://doi.org/10.3189/S0022143000002781) discussed [here](https://github.com/ESCOMP/CTSM/discussions/1960) and described for single point runs by [Dutch et al 2022](https://tc.copernicus.org/articles/16/4201/2022/).

### Specific notes

Contributors: Victoria Dutch, Leanne Wake

Testing in progress [here](https://github.com/NCAR/LMWG_dev/issues/4)

### CODE REVIEW

File: src/biogeophys/SoilTemperatureMod.F90 [hunk 1]
```diff
@@ -734,11 +734,16 @@ subroutine SoilThermProp (bounds, num_urbanc, filter_urbanc, num_nolakec, filter
                endif
             endif
 
-            ! Thermal conductivity of snow, which from Jordan (1991) pp. 18
+            ! Thermal conductivity of snow, which from Sturm (1997)
             ! Only examine levels from snl(c)+1 -> 0 where snl(c) < 1
             if (snl(c)+1 < 1 .AND. (j >= snl(c)+1) .AND. (j <= 0)) then  
-               bw(c,j) = (h2osoi_ice(c,j)+h2osoi_liq(c,j))/(frac_sno(c)*dz(c,j))
-               thk(c,j) = tkair + (7.75e-5_r8 *bw(c,j) + 1.105e-6_r8*bw(c,j)*bw(c,j))*(tkice-tkair)
+               bw(c,j) = ((h2osoi_ice(c,j)*1)+h2osoi_liq(c,j))/(frac_sno(c)*dz(c,j)) ! ==RHOS
```
Comment (wwieder) [inline src/biogeophys/SoilTemperatureMod.F90:740]: This line is actually is the same for both approaches.

File: src/biogeophys/SoilTemperatureMod.F90 [hunk 2]
```diff
@@ -47,7 +47,7 @@ module SoilTemperatureMod
   !     o The thermal conductivity of soil is computed from
   !       the algorithm of Johansen (as reported by Farouki 1981), and the
   !       conductivity of snow is from the formulation used in
-  !       SNTHERM (Jordan 1991).
+  !       Sturm (1997).
```
Comment (wwieder) [inline src/biogeophys/SoilTemperatureMod.F90:50]: Why is there so much documentation in the code itself here?  If we keep all the notes  (here and below) snow conductivity should reference both Jordan and Sturm papers.

File: src/biogeophys/SoilTemperatureMod.F90 [hunk 3]
```diff
@@ -734,11 +734,16 @@ subroutine SoilThermProp (bounds, num_urbanc, filter_urbanc, num_nolakec, filter
...
 
-            ! Thermal conductivity of snow, which from Jordan (1991) pp. 18
+            ! Thermal conductivity of snow, which from Sturm (1997)
             ! Only examine levels from snl(c)+1 -> 0 where snl(c) < 1
             if (snl(c)+1 < 1 .AND. (j >= snl(c)+1) .AND. (j <= 0)) then  
-               bw(c,j) = (h2osoi_ice(c,j)+h2osoi_liq(c,j))/(frac_sno(c)*dz(c,j))
-               thk(c,j) = tkair + (7.75e-5_r8 *bw(c,j) + 1.105e-6_r8*bw(c,j)*bw(c,j))*(tkice-tkair)
+               bw(c,j) = ((h2osoi_ice(c,j)*1)+h2osoi_liq(c,j))/(frac_sno(c)*dz(c,j)) ! ==RHOS
+               ! thk(c,j) = tkair + (7.75e-5_r8 *bw(c,j) + 1.105e-6_r8*bw(c,j)*bw(c,j))*(tkice-tkair) ! Original (Jordan) Parameterisation
+               if (bw(c,j)  <= 156) then !LMW or 0.156 ?
```
Comment (wwieder) [inline src/biogeophys/SoilTemperatureMod.F90:742]: Here is where I think we'll want to use a namelist option to chose between Jornand and Sturm formulations for `thk`

**Review by wwieder — COMMENTED** [review]
Thanks for contributing this @AdrienDams we're investigating the impact of the change to arctic soil temperatures and productivity now.  

Assuming the results seem promising, I suggest we add Jordan vs. Sturm namelist options for calculating snow thermal conductivity `thk`.  Updates for Sturm will also need to be added to the technote (e.q. 2.6.85)

**Comment by wwieder** [comment]
Thanks for your presentation yesterday @AdrienDams. Given results from your presentation and [tests](https://github.com/NCAR/LMWG_dev/discussions/10) done by @olyson  and @slevis-lmwg it seems like we should bring this in as a namelist option to let users chose between Jordan and Sturm snow conductivity formulations.  Adrien, can you add Keith and Sam as collaborators to your github PR so they can work on this PR and bring it to main?

File: src/biogeophys/SoilTemperatureMod.F90 (hunk 2 above)
Comment (slevis-lmwg) [reply src/biogeophys/SoilTemperatureMod.F90:50]: Half of the notes here and below are about the thermal conductivity of soil, so I will keep them and reference both papers.

File: src/biogeophys/SoilTemperatureMod.F90 (hunk 1 above)
Comment (slevis-lmwg) [reply src/biogeophys/SoilTemperatureMod.F90:740]: Returning line 740 to the original. Is the comment at the end `==RHOS` helpful to keep?

**Comment by slevis-lmwg** [comment]
Note to self:
For git push to this PR's branch from my branch, I'm using
`git push AdrienDams HEAD:Sturm`

File: src/biogeophys/SoilTemperatureMod.F90 [hunk 4]
```diff
@@ -734,11 +735,29 @@ subroutine SoilThermProp (bounds, num_urbanc, filter_urbanc, num_nolakec, filter
...
             ! Only examine levels from snl(c)+1 -> 0 where snl(c) < 1
             if (snl(c)+1 < 1 .AND. (j >= snl(c)+1) .AND. (j <= 0)) then  
                bw(c,j) = (h2osoi_ice(c,j)+h2osoi_liq(c,j))/(frac_sno(c)*dz(c,j))
-               thk(c,j) = tkair + (7.75e-5_r8 *bw(c,j) + 1.105e-6_r8*bw(c,j)*bw(c,j))*(tkice-tkair)
+!              TODO slevis: Add namelist option thermal_cond_snow and then
+!                           uncomment relevant lines below
+!              select case (thermal_cond_snow)
+!              case ('Jordan1991')
+!                 thk(c,j) = tkair + (7.75e-5_r8 *bw(c,j) + 1.105e-6_r8*bw(c,j)*bw(c,j))*(tkice-tkair)
+!              case ('Sturm1997')
```
Comment (slevis-lmwg) [inline src/biogeophys/SoilTemperatureMod.F90:747]: @wwieder 
I wanted to make sure I had the default value correct... Are keeping Jordan1991 as default or changing it to Sturm1997?

Comment (slevis-lmwg) [reply src/biogeophys/SoilTemperatureMod.F90:747]: ...and maybe the right answer is Jordan1991 for now to get bfb same answers and we change the default later in an answer changing tag.

**Comment by slevis-lmwg** [comment]
Test-suites
- cheyenne IN PROGRESS
- izumi

**Comment by slevis-lmwg** [comment]
Labeled as bfb because I made Jordan1991 the default option.

**Pull request closed without merging**